sql/src/main/java/io/crate/execution/engine/aggregation/impl/CollectSetAggregation.java

```diff
@@ -40,24 +41,33 @@
 import java.util.HashSet;
 import java.util.Set;
 
-public class CollectSetAggregation extends AggregationFunction<Set<Object>, Set<Object>> {
+public class CollectSetAggregation extends AggregationFunction<Set<Object>, Object[]> {
 
     public static final String NAME = "collect_set";
     private final SizeEstimator<Object> innerTypeEstimator;
 
-    private FunctionInfo info;
+    private final FunctionInfo info;
+    /**
+     * The partial return type is a SetType so we can maintain the java.util.Set intermediate type used for
```

Comment:
Not sure if this is the best place for this comment? The contract that the partialType is used for the intermediate state should imho be specified in `AggregationFunction`.  And the assignment to the mentioned set type isn't even happening here.

I'm also not sure if `returnType` is accurate. The partial data types are also used as inputs to the reduce function. `returnType` to me would imply a more narrow scope.

Reply:
Looking at the `AggregationFunction` I think the types and general functions documentation make this clear already, I'll drop this comment altogether.

Reply:
Thanks for the review and the suggestions. I'm quite happy with this cleanup and scope reduction for the SetType